pkg/upath/safejoin.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package upath
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// SafeFilepathJoin safely joins two paths path1+path2. The resulting path will
+// always be contained within path1 even if path2 tries to escape with "../".
+//
+// Note: This only works on Unix-like file systems.
```

Comment:
it would work on plan9 too but nobody calls that unix-like :-)
why would this not work on windows?

Reply:
On windows, they have paths like `C:\Documents\`, so prefixing with a `/` does not make sense.

Reply:
so should it have a build tag for ! windows?